### ISSUE TITLE
D3 Chart Issues & Questions

### ISSUE THREAD

**Issue by justinshreve** [woocommerce/wc-admin#320]
Well implementing the revenue report, I ran into a couple issues/questions.

**Issue 1 - Hour Intervals**

I can’t seem to find a way to pass hours accurately as data points. I tried something like 

```
[
    {
        "date":"2018-08-01T00:00:00+00:00",
        "Custom (Aug 1, 2018)":58929.99,
        "Previous Period (Jul 31, 2018)":160130.74000000002
    },
    {
        "date":"2018-08-01T12:00:00+00:00",
        "Custom (Aug 1, 2018)":3805.56,
        "Previous Period (Jul 31, 2018)":0
    },
   {
        "date":"2018-08-01T02:00:00+00:00",
        "Custom (Aug 1, 2018)":3805.56,
        "Previous Period (Jul 31, 2018)":0
    },
]
```

But the data points overflow the grid. 

<img width="946" alt="screen shot 2018-08-24 at 12 39 48 pm" src="https://user-images.githubusercontent.com/689165/44596613-d4b16d80-a79a-11e8-8eef-b3f911923478.png">

---

**Issue 2 - Axis off by 1**

The axis seems to be off by 1. See the screenshot below for an example. The tooltip shows the correct “August 1st”, but the axis shows a 31.

<img width="419" alt="screen shot 2018-08-24 at 11 28 20 am" src="https://user-images.githubusercontent.com/689165/44593645-fc500800-a791-11e8-9360-f3d54faaff2d.png">

This also happens on the static dashboard example.

---

**Issue 3 - No way to set tooltip with a formatted date**

For week/month/quarter intervals, the axis shows the correct gaps — but it would be nice if I could pass a formatted date to the tooltip such as “Week of ____” instead of just the first day of the week displaying.

<img width="275" alt="screen shot 2018-08-24 at 11 31 32 am" src="https://user-images.githubusercontent.com/689165/44593665-08d46080-a792-11e8-9659-1b6e910b1583.png">

cc @greenafrican @timmyc

**Comment by greenafrican** [comment]
@justinshreve 

Quick question regarding:
```
[
    {
        "date":"2018-08-01T00:00:00+00:00",
        "Custom (Aug 1, 2018)":58929.99,
        "Previous Period (Jul 31, 2018)":160130.74000000002
    },
    {
        "date":"2018-08-01T12:00:00+00:00",
        "Custom (Aug 1, 2018)":3805.56,
        "Previous Period (Jul 31, 2018)":0
    },
   {
        "date":"2018-08-01T02:00:00+00:00",
        "Custom (Aug 1, 2018)":3805.56,
        "Previous Period (Jul 31, 2018)":0
    },
]
```

Is this hourly data? Any reason it goes from `00` to `12` to `02`?

**Comment by justinshreve** [comment]
@greenafrican Yes that is hourly data.

There is a TODO in progress on the API side that would zero fill the missing entries. In this case only 00, 12, and 02 contain order data for that day.

**Comment by greenafrican** [comment]
@justinshreve Ok, yeah much better if that is done on the API. Is `02` = 2am? I'd expect the data to be ordered too (we can do this in the chart but it probably needs to be done for the tables, csv etc.)

**Comment by justinshreve** [comment]
@greenafrican Yup 2AM. It's possible I edited those examples by hand to try a couple different things, so it may have been ordered in the first place -- but I'll make sure that's done correctly.

**Comment by greenafrican** [comment]
@justinshreve no problem. I'm going to push a new PR up with patches to properly support custom x and y axis formats. It's working pretty well with hourly intervals and i'll test others too.

**Comment by greenafrican** [comment]
Closed in #360
